src/components/export-gist.jsx

```diff
@@ -0,0 +1,50 @@
+/* global XMLHttpRequest:false window:false */
+/* eslint-disable no-alert */
+import React from "react";
+
+export default class ExportGist extends React.Component {
+  parseJSON(responseText) {
+    try {
+      return JSON.parse(responseText);
+    } catch (e) {
+      window.alert("Unable to create Gist");
+    }
+    return null;
+  }
+
+  postGist() {
+    const {containerElement} = this.props;
+    const request = new XMLHttpRequest();
+    request.onreadystatechange = () => {
+      if (request.readyState === 4 && request.status === 201) {
+        const response = this.parseJSON(request.responseText);
+        window.open(response.html_url, "_blank");
+      }
+    };
+    const content = containerElement.getElementsByClassName("ecologyCode")[0];
```

Comment:
It would be better to pass in a reference to the actual `DOM` node you want to grab the `textContent` from. That way, the `ExportGist` component doesn't need to have knowledge of the internal HTML structure of the ecology container.